tools/indexscan/main.go

```diff
@@ -109,6 +110,81 @@ type searchStrategy interface {
 	Search(ctx context.Context, r *git.Repository, hashes []string) (closest []string, matched, total int, err error)
 }
 
+// DynamicTreeSearchStrategy  searches all TreeObjects in the repository for
+// the number of matches against the input files provided.
+type DynamicTreeSearchStrategy struct {
```

Comment:
i.e. "dynamic programming." it's annotating all trees (which, in git, are both "full" trees and "sub" trees) with the number of matches and coalescing that up to all commits' full trees.

Reply:
What about Memoized? Or maybe Cached?

IIRC "Dynamic" in dynamic programming was mostly added for showmanship rather than precision: "because it sounded impressive" and "to upstage Dantzig's linear programming by adding dynamic"

https://en.wikipedia.org/wiki/Dynamic_programming#:~:text=The%20word%20dynamic%20was%20chosen,and%20because%20it%20sounded%20impressive.

Reply:
idk no matter how unfortunate the naming is, it's descriptive and generally accepted. i see what you're saying (and I actually read through the wikipedia page to see whether there was a less obnoxious alternative but it is what is is)